services/src/main/java/org/keycloak/protocol/oidc/mappers/OIDCAttributeMapperHelper.java

```diff
@@ -70,29 +69,78 @@ public static Object mapAttributeValue(ProtocolMapperModel mappingModel, Object
-            return attributeValue.toString();
-        } else if (type.equals("long")) {
-            if (attributeValue instanceof Long) return attributeValue;
-            if (attributeValue instanceof String) return Long.valueOf((String)attributeValue);
-            throw new RuntimeException("cannot map type for token claim");
-        } else if (type.equals("int")) {
-            if (attributeValue instanceof Integer) return attributeValue;
-            if (attributeValue instanceof String) return Integer.valueOf((String)attributeValue);
-            throw new RuntimeException("cannot map type for token claim");
+        switch (type) {
+            case "boolean":
+                Object booleanObject = getBoolean(attributeValue);
+                if (booleanObject != null) return booleanObject;
+                if (attributeValue instanceof List) {
+                    return transform((List<Object>) attributeValue, OIDCAttributeMapperHelper::getBoolean);
+                }
+                throw new RuntimeException("cannot map type for token claim");
+            case "String":
+                if (attributeValue instanceof String) return attributeValue;
+                if (attributeValue instanceof List) {
+                    return transform((List<String>) attributeValue, OIDCAttributeMapperHelper::getString);
+                }
+                return attributeValue.toString();
+            case "long":
+                Object longObject = getLong(attributeValue);
+                if (longObject != null) return longObject;
+                if (attributeValue instanceof List) {
+                    return transform((List<Long>) attributeValue, OIDCAttributeMapperHelper::getLong);
+                }
+                throw new RuntimeException("cannot map type for token claim");
+            case "int":
+                Object intObject = getInteger(attributeValue);
+                if (intObject != null) return intObject;
+                if (attributeValue instanceof List) {
+                    return transform((List<Integer>) attributeValue, OIDCAttributeMapperHelper::getInteger);
+                }
+                throw new RuntimeException("cannot map type for token claim");
+            default:
+                return null;
         }
-        return attributeValue;
+    }
+
+    private static String getString(Object attributeValue) {
+        return attributeValue.toString();
+    }
+
+
+    private static Long getLong(Object attributeValue) {
+        if (attributeValue instanceof Long) return (Long) attributeValue;
+        if (attributeValue instanceof String) return Long.valueOf((String) attributeValue);
+        return null;
+    }
+
+    private static Integer getInteger(Object attributeValue) {
+        if (attributeValue instanceof Integer) return (Integer) attributeValue;
+        if (attributeValue instanceof String) return Integer.valueOf((String) attributeValue);
```

Comment:
`NumberFormatException` has to be caught and converted to `RuntimeException` when exception occurs.